### ISSUE TITLE
Next@9 compatibily 

### ISSUE THREAD

**Issue by dkrefta** [preactjs/preact#3619]
Hey Guys, I'm trying to check which version this old legacy Next JS project can use the preact, there is nothing that I can find about compatibility? the NextJS version of this old project is 9 ( unfortunately I just want to do a little bit of booster in performance until we finish the another new project lol )

**Additional context (optional)**

**Comment by rschristian** [comment]
Are you asking about Preact's NextJS plugin ([`next-plugin-preact`](https://github.com/preactjs/next-plugin-preact)), or Preact itself?

It looks like the v9.0.0 release of Next supports React 16.x, so there shouldn't be any issues there (theoretically) in swapping in `preact/compat`. Compatibility in this case is mostly up to Next; while Preact offers nearly full compat with React, Next still needs to facilitate the aliasing.

**Comment by JoviDeCroock** [comment]
Yes, Next@9 should be fully supported. It's a bit old so I'm not entirely sure whether the `next-plugin-preact` does too much or too little but following the steps that we do there should be enough to make it work.

- replace your react dependencies with the stub one from preact
- maybe add `prefresh` if you need `fast-refresh` no idea if that was supported in v9
